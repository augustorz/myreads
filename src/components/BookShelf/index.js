import React from 'react';
import PropTypes from 'prop-types';

import Book from '../Book';

const BookShelf = ({ title, books, onChangeBookShelf }) => (
  <div className="bookshelf">
    <h2 className="bookshelf-title">{title}</h2>
    <div className="bookshelf-books">
      <ol className="books-grid">
        {books.map(book => (
          <Book key={book.id} book={book} onChangeBookShelf={onChangeBookShelf} />
        ))}
      </ol>
    </div>
  </div>
);

BookShelf.propTypes = {
  title: PropTypes.string.isRequired,
  books: PropTypes.instanceOf(Array),
  onChangeBookShelf: PropTypes.func.isRequired,
};

BookShelf.defaultProps = {
  books: [],
};

export default BookShelf;
