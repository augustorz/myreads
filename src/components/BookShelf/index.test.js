import React from 'react';
import { shallow } from 'enzyme';
import createBooks from '../../mocks/books';

import BookShelf from '.';

describe('BookShelf', () => {
  let wrapper;

  describe('when books is present', () => {
    beforeEach(() => {
      const books = createBooks({ quantity: 4 });

      wrapper = shallow(
        <BookShelf
          title="read"
          books={books}
          onChangeBookShelf={jest.fn()}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when books is not present', () => {
    beforeEach(() => {
      wrapper = shallow(
        <BookShelf
          title="currentlyReading"
          onChangeBookShelf={jest.fn()}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
