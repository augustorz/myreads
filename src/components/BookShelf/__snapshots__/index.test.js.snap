// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf when books is not present should match snapshot 1`] = `"<div class=\\"bookshelf\\"><h2 class=\\"bookshelf-title\\">currentlyReading</h2><div class=\\"bookshelf-books\\"><p class=\\"bookshelf-message\\">No books found.</p></div></div>"`;

exports[`BookShelf when books is present should match snapshot 1`] = `"<div class=\\"bookshelf\\"><h2 class=\\"bookshelf-title\\">read</h2><div class=\\"bookshelf-books\\"><ol class=\\"books-grid\\"><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width:128px;height:193px;background-image:url(&quot;thumbnail_0.png&quot;)\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option selected=\\"\\" value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 0</div><div class=\\"book-authors\\">Title Mock 0</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width:128px;height:193px;background-image:url(&quot;thumbnail_1.png&quot;)\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option selected=\\"\\" value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 1</div><div class=\\"book-authors\\">Title Mock 1</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width:128px;height:193px;background-image:url(&quot;thumbnail_2.png&quot;)\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option selected=\\"\\" value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 2</div><div class=\\"book-authors\\">Title Mock 2</div></div></li><li><div class=\\"book\\"><div class=\\"book-top\\"><div class=\\"book-cover\\" style=\\"width:128px;height:193px;background-image:url(&quot;thumbnail_3.png&quot;)\\"></div><div class=\\"book-shelf-changer\\"><select><option value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option selected=\\"\\" value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div></div><div class=\\"book-title\\">Author Mock 3</div><div class=\\"book-authors\\">Title Mock 3</div></div></li></ol></div></div>"`;
